packages/amplify-provider-awscloudformation/src/configuration-manager.js

```diff
@@ -509,18 +541,81 @@ function loadConfigFromPath(context, profilePath) {
   throw new Error(`Invalid config ${profilePath}`);
 }
 
-async function loadConfigurationForEnv(context, env) {
+async function getAdminCredentials(idToken, identityId, region) {
+  const cognitoIdentity = new aws.CognitoIdentity({ region });
+  const login = idToken.payload.iss.replace('https://', '');
+  return cognitoIdentity
+    .getCredentialsForIdentity({
+      IdentityId: identityId,
+      Logins: {
+        [login]: idToken.jwtToken,
+      },
+    })
+    .promise();
+}
+
+async function loadConfigurationForEnv(context, env, appId) {
   const projectConfigInfo = getConfigForEnv(context, env);
-  if (projectConfigInfo.configLevel === 'project') {
+  const { print } = context;
+  let awsConfig;
+  if (projectConfigInfo.configLevel === 'amplifyAdmin' || appId) {
+    projectConfigInfo.configLevel = 'amplifyAdmin';
+    appId = appId || stateManager.getMeta().providers.awscloudformation.AmplifyAppId;
+
+    if (!doAdminCredentialsExist(appId)) {
+      print.info('');
+      print.error(`No credentials found for appId: ${appId}`);
+      print.info(`If the appId is correct, try running amplify configure --appId ${appId}`);
+      process.exit(1);
+    }
+
+    try {
+      const authConfig = await getRefreshedTokens(appId, context.printappId, print);
+      const { idToken, IdentityId, region } = authConfig;
+      // use tokens to get creds and assign to config
+      let credentials = (await getAdminCredentials(idToken, IdentityId, region)).Credentials;
+
+      awsConfig = {
+        accessKeyId: credentials.AccessKeyId,
+        expiration: credentials.Expiration,
+        region,
+        secretAccessKey: credentials.SecretKey,
+        sessionToken: credentials.SessionToken,
+      };
+
+      aws.config.update(awsConfig);
+
+      const sts = new aws.STS();
+      credentials = (
+        await sts
+          .assumeRole({
+            RoleArn: idToken.payload['cognito:preferred_role'],
+            RoleSessionName: 'amplifyadmin',
```

Comment:
Shouldn't `RoleSessionName` be a unique identifier for the session? It could help with troubleshooting and logging and such. https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html

Reply:
Not a requirement to make this unique. Right now the customer would be able to know that the role is assumed using the CLI with the above role session name.